tokens/tokens.md

```diff
@@ -0,0 +1,207 @@
+executed if a call to spend returns true. (TODO: Decide how this plays
+with contract calls that tries to pass tokens as value, etc).
+
+Note that the `final` field in the ANT takes precedence over the
+governing contract. If the ANT is final, no minting can occur.  (TODO:
+Perhaps the contract should have a `finalize` endpoint as well?)
+
+## ANT transactions
+
+### ANT create transaction (`ant_create_tx`)
+
+The ANT create transaction takes the argument:
+- Meta data : string
+- Amount
+- Recipient
+- Final
+- Contract
+- Parent  (TODO: Hierarchical tokens?)
+
+The `amount` is the number of tokens to mint at create time. Set to
+`0` if none should be minted. This can for example be combined with
+setting the `final` argument to true, thereby immediately minting all
+tokens that will ever exist.
+
+The `recipient` is the recipient of the minted tokens in `amount`. If
+not provided, the tokens are given to the creator's account.
+
+The `parent` is a pointer to a parent token for hierarchical
+tokens. (TODO: hierarchical tokens?)
+
+### ANT finalize transaction (`ant_finalize_tx`)
+
+The ANT finalize transaction takes the arguments:
+- owner
+- ANT
+
+The finalize transaction can only be submitted by the actual owner,
+and only if the ANT is not already finalized.
+
+### ANT mint transaction (`ant_mint_tx`)
+The ANT mint transaction takes the argument:
+- owner
+- ANT
+- amount
+- recipient
+- final
+
+Only the `owner` can mint new tokens, but it can pass the minted
+tokens to a recipient. If `final` is set to true, the ANT will be
+finalized after the new tokens are minted.
+
+### ANT destroy transaction
+TODO: Should we be able to destroy an ANT that has a `total_supply` of 0?
+
+
+## Token transactions
+
+### Token trade transaction
+The token trade transaction takes the argument:
+- trades
+
+The `trades` field contans a non-empty list of token transfers. A
+single token transfer consists of the fields:
+
+- Sender
+- Receiver
+- Amount
+- An ANT id
+
+This construct makes it possible to atomically perform complicated
+trade operations involving more than one ANT and also aeons (TODO: How
+to signal that a trade concerns aeons) between multiple parties.
+
+The transaction must be signed by all senders. (TODO: Should we use
+some alternative multisig format here?).
+
+
+### Token burn transaction
+
+The token burn transaction contains the fields
+- Account (owner of the tokens, not necessarily of the ANT)
+- Amount
+- ANT
+
+Destroy an `Amount` of `ANT` tokens currently owned by the
+`account`. The burned amount is also counted from the `total_supply`
+in the ANT object.
+
+### Other transactions on tokens
+
+Tokens can also be transfered from one account to another through any
+other transaction that can pass an amount. You can:
+- spend tokens through the `spend_tx`
+- pass tokens as value in a `contract_call_tx` or `contract_create_tx`
+- pass tokens as value in contract calls in a smart contract.
```

Comment:
Consider allowing multiple currencies (for instance, some tokens + a tip in aeons).